File: nest/backend-ms1/src/app/module/admin/service/achat/AchatItemAdminServiceImpl.ts
import {Injectable, NotFoundException} from "@nestjs/common";

import {AbstractServiceImpl} from "src/app/zynerator/service/AbstractServiceImpl";

import {AchatItemDao} from "src/app/controller/dao/facade/core/achat/AchatItemDao";
import {AchatItemCriteria} from "src/app/controller/dao/criteria/core/achat/AchatItemCriteria";
import {AchatItem} from "src/app/controller/bean/core/achat/AchatItem";
import {AchatItemDto} from "src/app/controller/dto/achat/AchatItemDto";


@Injectable()
export class AchatItemAdminServiceImpl extends AbstractServiceImpl<AchatItem, AchatItemCriteria, AchatItemDao>{

    constructor(private readonly dao: AchatItemDao ,
    ) {
        super(dao);
    }

    async save(item: AchatItem): Promise<AchatItem> {
        const saved = await this.dao.save(item);
        return saved;
    }


    async update(item: AchatItem): Promise<AchatItem> {
        const saved = await this.dao.saveOrUpdate(item);
        return saved;
    }

    async updateMultiple(items: AchatItem[]): Promise<void> {
        if (items) {
            items.forEach(e => this.update(e))
        }
    }

    async  findAllOptimized(): Promise<AchatItemDto[]> {
        return this.dao.findAllOptimized();
    }

    async findAll(): Promise<AchatItem[]> {
        return this.dao.findAll();
    }

    async findById(id: number): Promise<AchatItem> {
        return this.dao.findById(id);
    }


    async deleteById(id: number): Promise<number> {
        const existing = await this.findById(id);
        if (!existing) {
            throw new NotFoundException(`Achat item with ID ${id} not found.`);
        }
        const result = await this.dao.deleteById(existing.id);
        return result;
    }

    async deleteMultiple(items: AchatItem[]): Promise<AchatItem[]> {
        const deletedItems: AchatItem[] = [];
        for (const item of items) {
            await this.deleteById(item.id);
            deletedItems.push(item);
        }
        return deletedItems;
    }

    async findByProduitId(id: number): Promise<AchatItem[]> {
        return this.dao.findByProduitId(id);
    }

    async deleteByProduitId(id: number): Promise<number> {
        return this.dao.deleteByProduitId(id);
    }
    async findByAchatId(id: number): Promise<AchatItem[]> {
        return this.dao.findByAchatId(id);
    }

    async deleteByAchatId(id: number): Promise<number> {
        return this.dao.deleteByAchatId(id);
    }

    async findWithAssociatedLists(id: number): Promise<AchatItem> {
        const result = await this.dao.findById(id);
    return result;
    }

    async updateWithAssociatedLists(item: AchatItem): Promise<AchatItem> {
         return await this.dao.saveOrUpdate(item);
    }
}

File: spring-next/v2/backend-ms1/src/main/java/ma/zs/stocky/service/impl/admin/achat/AchatItemAdminServiceImpl.java
package ma.zs.stocky.service.impl.admin.achat;


import ma.zs.stocky.bean.core.achat.AchatItem;
import ma.zs.stocky.dao.criteria.core.achat.AchatItemCriteria;
import ma.zs.stocky.dao.facade.core.achat.AchatItemDao;
import ma.zs.stocky.dao.specification.core.achat.AchatItemSpecification;
import ma.zs.stocky.service.facade.admin.achat.AchatItemAdminService;
import ma.zs.stocky.zynerator.service.AbstractServiceImpl;
import ma.zs.stocky.zynerator.util.ListUtil;
import org.springframework.stereotype.Service;
import java.util.List;
import java.util.ArrayList;





import org.springframework.beans.factory.annotation.Autowired;

import ma.zs.stocky.service.facade.admin.achat.AchatAdminService ;
import ma.zs.stocky.bean.core.achat.Achat ;
import ma.zs.stocky.service.facade.admin.commun.ProduitAdminService ;
import ma.zs.stocky.bean.core.commun.Produit ;

import java.util.List;
@Service
public class AchatItemAdminServiceImpl extends AbstractServiceImpl<AchatItem, AchatItemCriteria, AchatItemDao> implements AchatItemAdminService {







    public List<AchatItem> findByProduitId(Long id){
        return dao.findByProduitId(id);
    }
    public int deleteByProduitId(Long id){
        return dao.deleteByProduitId(id);
    }
    public long countByProduitReference(String reference){
        return dao.countByProduitReference(reference);
    }
    public List<AchatItem> findByAchatId(Long id){
        return dao.findByAchatId(id);
    }
    public int deleteByAchatId(Long id){
        return dao.deleteByAchatId(id);
    }
    public long countByAchatReference(String reference){
        return dao.countByAchatReference(reference);
    }






    public void configure() {
        super.configure(AchatItem.class, AchatItemSpecification.class);
    }


    @Autowired
    private AchatAdminService achatService ;
    @Autowired
    private ProduitAdminService produitService ;

    public AchatItemAdminServiceImpl(AchatItemDao dao) {
        super(dao);
    }

}